crowbar_framework/lib/crowbar/upgrade_status.rb

```diff
@@ -183,10 +198,36 @@ def save_openstack_backup(backup_location)
       end
     end
 
-    def save_upgrade_mode(mode)
+    def save_suggested_upgrade_mode(mode)
       ::Crowbar::Lock::LocalBlocking.with_lock(shared: false, logger: @logger, path: lock_path) do
         load_while_locked
-        progress[:upgrade_mode] = mode
+        progress[:suggested_upgrade_mode] = mode
+        # reset the selected_upgrade_mode if it the current selection is impossible
+        # i.e. non_disruptive is selected, but only :normal is possible
+        if [:normal, :none].include? mode
+          progress[:selected_upgrade_mode] = nil
+        end
+        save
+      end
+    end
+
+    def save_selected_upgrade_mode(mode)
+      ::Crowbar::Lock::LocalBlocking.with_lock(shared: false, logger: @logger, path: lock_path) do
+        load_while_locked
+        # It's ok to change the upgrade mode until starting the services step
+        unless pending? :services
```

Comment:
This part is actually not needed on Cloud6 side (but probably better anyway for consistency...)